src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -114,42 +116,49 @@ public synchronized void init() {
     new Thread(new Runnable() {
       @Override
       public void run() {
-        String targetUser = mesosCloud.getSlavesUser();
-        String webUrl = Jenkins.getInstance().getRootUrl();
-        if (webUrl == null) webUrl = System.getenv("JENKINS_URL");
-        // Have Mesos fill in the current user.
-        FrameworkInfo framework = FrameworkInfo.newBuilder()
-          .setUser(targetUser == null ? "" : targetUser)
-          .setName(mesosCloud.getFrameworkName())
-          .setPrincipal(mesosCloud.getPrincipal())
-          .setCheckpoint(mesosCloud.isCheckpoint())
-          .setWebuiUrl(webUrl != null ? webUrl :  "")
-          .build();
-
-        LOGGER.info("Initializing the Mesos driver with options"
-        + "\n" + "Framework Name: " + framework.getName()
-        + "\n" + "Principal: " + framework.getPrincipal()
-        + "\n" + "Checkpointing: " + framework.getCheckpoint()
-        );
-
-        if (StringUtils.isNotBlank(mesosCloud.getSecret())) {
+        try {
+          String targetUser = mesosCloud.getSlavesUser();
+          String webUrl = Jenkins.getInstance().getRootUrl();
+          if (webUrl == null) webUrl = System.getenv("JENKINS_URL");
+          StandardUsernamePasswordCredentials credentials = mesosCloud.getCredentials();
+          String principal = credentials == null ? "" : credentials.getUsername();
```

Comment:
Previously, the default principal was "jenkins". Now it is being changed to empty string. This is not a backwards compatible change!